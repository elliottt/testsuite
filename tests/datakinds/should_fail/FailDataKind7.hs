{-# LANGUAGE DataKinds #-}
{-# LANGUAGE PolyKinds #-}
{-# LANGUAGE KindSignatures #-}
module FailDataKind7 where

-- unholy recursion
data kind K = T Foo
data Foo (a :: K) = Bar
