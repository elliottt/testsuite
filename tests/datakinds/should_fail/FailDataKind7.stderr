
DataKind7.hs:8:16:
    Type constructor ‛K’ cannot be used here
      (it is defined and used in the same recursive group)
    In the kind ‛K’
